app/src/main/java/com/imdb_compose/MainActivity.kt

```diff
@@ -1,47 +1,1195 @@
+                    movie.poster_path
+                }
+            )
+            // ribbon
+            Ribbon(paddingStart = 12.dp)
+        }
+        // details box
+        Column (
+            modifier = Modifier
+                .width(288.dp)
+                .shadow(
+                    color = gray500,
+                    offsetX = 12.dp,
+                    offsetY = 0.dp,
+                    blurRadius = 4.dp,
+                    blurRadiusFilter = "SOLID"
+                )
+        ) {
+            Box(modifier = Modifier.padding(start = 16.dp, top = 8.dp), contentAlignment = Alignment.CenterStart) {
+                Details(
+                    catagory = catagory,
+                    rank = "",
+                    movie = movie,
+                    show = null,
+                    person = null,
+                    showCircleI = false
+                )
+            }
+        }
+    }
+}
+
+@Composable
+fun Details(
+    catagory: String,
+    rank: String,
+    movie: MovieResult?,
+    show: TvResult?,
+    person: ActorResult?,
+    showCircleI: Boolean
+) {
+    Box (modifier = Modifier.width(180.dp)) {
+        Column (
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(start = 4.dp, bottom = 8.dp),
+            verticalArrangement = Arrangement.SpaceEvenly
+        ) {
+            // Trending
+            if (Regex("trending").containsMatchIn(catagory.lowercase())) {
+                Box(
+                    modifier = Modifier.padding(start = 4.dp),
+                    contentAlignment = Alignment.CenterStart
+                ) {
+                    Text(
+                        text = rank,
+                        fontSize = MaterialTheme.typography.displaySmall.fontSize,
+                        fontFamily = MaterialTheme.typography.displaySmall.fontFamily,
+                        fontStyle = MaterialTheme.typography.displaySmall.fontStyle,
+                        fontWeight = MaterialTheme.typography.displaySmall.fontWeight
                     )
                 }
             }
+            // person name
+            if (person != null) {
+                Text(
+                    text = person.original_name,
+                    modifier = Modifier.padding(start = 8.dp),
+                    overflow = TextOverflow.Ellipsis,
+                    softWrap = false
+                )
+                Text(
+                    text = person.known_for_department,
+                    modifier = Modifier.padding(start = 8.dp)
+                )
+            }
+            // rating
+            Row(modifier = Modifier.fillMaxWidth().padding(bottom = 8.dp)) {
+                Icon(
+                    modifier = Modifier.padding(start = 8.dp),
+                    imageVector = Icons.Filled.Star,
+                    contentDescription = "rating",
+                    tint = ripeMango
+                )
+                if (movie != null) {
```

Comment:
As a rule of thumb, if you see yourself doing this chain of null checks, take it as a code smell you should avoid. Have one parameter to this function that asks for the one piece of data you want and let the caller worry about pulling the the relevant info.